### DISCUSSION TITLE
Compilation terminated while building the test problems on the Moose installation guide website

### DISCUSSION THREAD

**Discussion by troyNCSU** [idaholab/moose#18208] [Q&A Getting Started] [answered]
Hi,
I have encountered a problem while I try to build the testing problem mentioned on the [getting started installation](https://mooseframework.inl.gov/getting_started/installation/conda.html) page. The following error messages show up on the terminal window. If you could help me in this regard, that will be great. 
Thanks,
Tuhin 
**Error messages:**
ake: execvp: python-config: Permission denied
make: execvp: python-config: Permission denied
/projects/moose/moosetools/contrib/hit/hit.cpp:4:10: fatal error: Python.h: No such file or directory
    4 | #include "Python.h"
      |          ^~~~~~~~~~

**Comment by GiudGiud** [comment]
Hello

Are you following the Conda install path?
Which operating system is that for?

The permission denied error is odd too. 

Guillaume

**Reply by GiudGiud** [reply]
Actually @troyNCSU could you please let us know how you solved the problem in this thread? This will help future WSL users and will help me help them. Even if it was something super simple, it s very valuable

**Reply by GiudGiud** [reply]
Or was it just changing the working directory?

**Comment by troyNCSU** [comment]
Hi Guillaume,

Thank you for the quick response. I just followed the instructions
mentioned on the getting started webpage on the Ubuntu terminal window. I
have not changed anything for the Conda install path.

The base operating system is Windows-10, but I have installed Ubuntu 20.04
WSL and used this for the Moose installation.

In addition, I just want to mention that I also followed the instructions
mentioned under the Build Issue on the Moose Troubleshooting page
<https://mooseframework.inl.gov/help/troubleshooting.html#condaissues>. All
the steps there were carried out successfully.

Thanks,
Tuhin

**Reply by GiudGiud** [reply]
Ok that's good. 
Just checking your environment:
what does `echo $LIBMESH_DIR; echo $MOOSE_DIR` return ?
what does `which python` return?
what does `ls -l` return? (do you have permission to write in the folder you're working in)
Can you locate the python.h header on your system? You can use `locate Python.h`

**Reply by troyNCSU** [reply]
Hi Guillaume,

Here are the answers to your questions:

echo $LIBMESH_DIR
/home/$user/miniconda3/libmesh

echo $MOOSE_DIR
... returns blank! ...


which python
/home/$user/miniconda3/bin/python

 ls -l
-rw-r--r-- 1 $user $user 94235922 Jun 27 23:43 Miniconda3-latest-Linux-x86_64.sh
drwxr-xr-x 1 $user $user      512 Jun 29 17:36 miniconda3
drwxr-xr-x 1 $user $user      512 Jun 27 23:54 projects

 locate Python.h
/home/$user/miniconda3/include/python3.8/Python.h
/home/$user/miniconda3/pkgs/python-3.8.5-h7579374_1/include/python3.8/Python.h

Thanks,
Tuhin

**Reply by GiudGiud** [reply]
Well looks like you have permissions in projects, and that the header reported missing is available.
So maybe the wrong compiler is used? That would be odd because it looks like your environment is well set up. 

What do `echo $CXX` and `which mpicxx` return ?
$user is an odd username. I wonder if that's the issue.

**Reply by troyNCSU** [reply]
Hi Guillaume,

I would like to inform you that the problem has got resolved. The working directory has been updated and now the test runs are running perfectly. Thanks for helping me out, I really appreciate it. 

Thanks,
Tuhin

**Reply by GiudGiud** [reply]
Glad to hear it. Thanks for letting us know

Guillaume

**Comment by troyNCSU** [comment]
How exactly the problem has been resolved is still mysterious. Earlier the
Python.h file was not accessible, after running several commands as you
suggested in the earlier posts, it looks like the file has got accessible
and the test problem build has done successfully.

**Comment by lynnmunday** [comment, marked as answer]
We were able to resolve this problem with Python.h, building MOOSE and getting MPI to work.  They didn't have a symbolic link between python and python3 in their moose conda env bin.  They deleted that conda env and  reinstalled it with moose-tools and moose-libmesh and the link is now there and everything is working.  Their previous moose conda env was from October so maybe this link between python and python3 was recently added.  They were using WSL ubuntu which doesn't have python defined at all, it only has python3 and there is no symbolic link between the two.

**Reply by GiudGiud** [reply]
Great thanks for reporting this! Marking this as the answer since we did not really find a reason for why it had started working earlier, except that the procedure had been re-done.
